### PR TITLE
Feature/FOUR-14120: Backend - Fire job when task is assigned to a user

### PR DESCRIPTION
**Pull request by CarliPinell**
## Solution
- A Listener for the ActivityAssigned event in ProcessMaker/Providers/ProcessMakerServiceProvider.php was added in registerEvents method
- A Dispatch method for SmartInbox job was added.

## How to Test
- Create Inbox Rules
- Start a new request

## Related Tickets & Packages
https://processmaker.atlassian.net/browse/FOUR-14120

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.
ci:deploy

### CODE REVIEW

**Review by nolanpro — CHANGES_REQUESTED** [review]

File: ProcessMaker/Providers/ProcessMakerServiceProvider.php [hunk 1]
```diff
@@ -187,6 +189,13 @@ protected static function registerEvents(): void
 
             Facades\Log::debug('Broadcasting Notification ' . $event->broadcastType() . 'on channel(s) ' . $channels);
         });
+
+        //Fire job when task is assigned to a user
+        Facades\Event::listen(ActivityAssigned::class, function ($event) {
+            $processRequestToken = $event->getProcessRequestToken();
+            // Dispatch the SmartInbox job with the processRequestToken as parameter
+            SmartInbox::dispatch($processRequestToken);
```
[the remark this answers is not in the document]
Comment (CarliPinell) [reply ProcessMaker/Providers/ProcessMakerServiceProvider.php:197]: Solved.

File: ProcessMaker/Jobs/SmartInbox.php [hunk 2]
```diff
@@ -8,34 +8,31 @@
...
 {
     use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
 
-    public $task;
+    public $incomingTask;
     /**
      * Create a new job instance.
      */
-    public function __construct(ProcessRequestToken $task)
+    public function __construct(ProcessRequestToken $incomingTask)
```
[the remark this answers is not in the document]
Comment (CarliPinell) [reply ProcessMaker/Jobs/SmartInbox.php:23]: Solved.

File: ProcessMaker/InboxRules/ApplyAction.php [hunk 3]
```diff
@@ -2,6 +2,61 @@
...
+use Facades\ProcessMaker\InboxRules\MatchingTasks;
+use Illuminate\Support\Facades\Auth;
+use ProcessMaker\Facades\WorkflowManager;
+use ProcessMaker\Models\ProcessRequestToken;
+use ProcessMaker\SanitizeHelper;
+use ProcessMaker\Http\Resources\Task as Resource;
+
 class ApplyAction
 {
+    public function applyActionOnTask(ProcessRequestToken $task)
```
[the remark this answers is not in the document]
Comment (CarliPinell) [reply ProcessMaker/InboxRules/ApplyAction.php:13]: in Progress.

**Comment by boliviacoca** [comment]
Closing

**Pull request closed without merging**